pyth-sdk/src/lib.rs

```diff
@@ -134,134 +103,39 @@ impl Default for PriceStatus {
-    prev_publish_time:      UnixTimestamp,
+    pub id:    PriceIdentifier,
+    /// Price.
+    price:     Price,
+    /// Exponentially-weighted moving average (EMA) price.
+    ema_price: Price,
 }
 
 impl PriceFeed {
     /// Constructs a new Price Feed
     #[allow(clippy::too_many_arguments)]
-    pub fn new(
-        id: PriceIdentifier,
-        status: PriceStatus,
-        publish_time: UnixTimestamp,
-        expo: i32,
-        max_num_publishers: u32,
-        num_publishers: u32,
-        product_id: ProductIdentifier,
-        price: i64,
-        conf: u64,
-        ema_price: i64,
-        ema_conf: u64,
-        prev_price: i64,
-        prev_conf: u64,
-        prev_publish_time: UnixTimestamp,
-    ) -> PriceFeed {
+    pub fn new(id: PriceIdentifier, price: Price, ema_price: Price) -> PriceFeed {
         PriceFeed {
             id,
-            status,
-            publish_time,
-            expo,
-            max_num_publishers,
-            num_publishers,
-            product_id,
             price,
-            conf,
             ema_price,
-            ema_conf,
-            prev_price,
-            prev_conf,
-            prev_publish_time,
         }
     }
 
-    /// Get the current price and confidence interval as fixed-point numbers of the form a *
-    /// 10^e.
-    ///
-    /// Returns a struct containing the current price, confidence interval, and the exponent for
-    /// both numbers. Returns `None` if price information is currently unavailable for any
-    /// reason.
-    pub fn get_current_price(&self) -> Option<Price> {
-        if !matches!(self.status, PriceStatus::Trading) {
-            None
-        } else {
-            Some(Price {
-                price: self.price,
-                conf:  self.conf,
-                expo:  self.expo,
-            })
-        }
-    }
 
-    /// Get the "unchecked" current price and confidence interval as fixed-point numbers of the form
-    /// a * 10^e.
+    /// Get the "unchecked" price and confidence interval as fixed-point numbers of the form
+    /// a * 10^e along with its publish time.
     ///
-    /// Returns a struct containing the current price, confidence interval, and the exponent for
-    /// both numbers. This method returns the price value without checking availability of the
-    /// price. This value might not be valid or updated when the price is not available.
-    /// Please use `get_current_price` where possible.
-    pub fn get_current_price_unchecked(&self) -> Price {
-        Price {
-            price: self.price,
-            conf:  self.conf,
-            expo:  self.expo,
-        }
+    /// Returns a `Price` struct containing the current price, confidence interval, and the exponent
+    /// for both numbers, and publish time. This method returns the price value without checking
+    /// availability of the price. This value might not be valid or updated when the price is
```

Comment:
the part about availability is not right anymore. i think you just need to say that the price is as of the last update, which may be from arbitrarily far in the past, and the caller should probably check the timestamp before using it. Would also suggest leaving a pointer to the safe versions of this function below.

Reply:
I wanted to make Tom change so I did this to.